cgp/individual.py

```diff
@@ -1,43 +1,84 @@
+import numpy as np
+from typing import Callable, List, Union
+
+try:
+    import sympy  # noqa: F401
+    from sympy.core import expr as sympy_expr  # noqa: F401
+
+    sympy_available = True
+except ModuleNotFoundError:
+    sympy_available = False
+
+try:
+    import torch  # noqa: F401
+
+    torch_available = True
+except ModuleNotFoundError:
+    torch_available = False
+
 from .genome import Genome
 from .cartesian_graph import CartesianGraph
 
 
-class Individual:
-    """An individual representing a particular computational graph.
+def mutate_genome(genome: Genome, mutation_rate: float, rng: np.random.RandomState) -> bool:
+    """Mutate a given genome.
+
+    Parameters
+    ----------
+    mutation_rate : float
+        Proportion of genes to be mutated, between 0 and 1.
+    rng : numpy.RandomState
+        Random number generator instance.
+
+    Returns
+    -------
+    bool
+        Whether all mutations were silent.
     """
+    n_mutations = int(mutation_rate * len(genome.dna))
+    assert n_mutations > 0
+
+    graph = CartesianGraph(genome)
+    active_regions = graph.determine_active_regions()
+    only_silent_mutations = genome.mutate(n_mutations, active_regions, rng)
+
+    return only_silent_mutations
 
-    def __init__(self, fitness, genome):
+
+class IndividualBase:
+    """Base class for all individuals.
+    """
+
+    def __init__(self, fitness: Union[float, None], genome: Union[Genome, List[Genome]]) -> None:
```

Comment:
to avoid the unions for genome, wouldn't it be cleaner to only define this member in the derived classes?

Reply:
this is needed by the `clone` function.